### ISSUE TITLE
[prefer-includes] Add option to ignore simple regex

### ISSUE THREAD

**Issue by bschlenk** [typescript-eslint/typescript-eslint#3825]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

I was surprised to find that the auto fix for this rule would change a simple regex into a string.includes call. Sometimes the regex can be simpler to read, especially if it is surrounded by other regex tests that happen to use special regex characters.

Would you be open to adding an option to this rule to ignore simple regex?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.30.0` |
| `@typescript-eslint/parser`        | `4.30.0` |
| `TypeScript`                       | `4.4.2` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.3` |

**Comment by bradzacher** [comment]
Stylistic lint rules exist to enforce consistency. Adding special options to ignore things in specific ways instead breeds inconsistencies. This is often surprising to developers and seems like a false negative instead of intentional behaviour, harming trust into he linting ecosystem.

Additionally - your definition of "simple" might not align with another's. Which makes such an option difficult to implement in a way that won't be surprising to someone.

Finally, stylistic lint rules will never fit everyone's preferences, as is the nature of code style. We tend toward keeping the rules clamped down options-wise unless there's a really good reason for adding an option.

For these reasons, we are going to reject your proposal. Thanks for taking the time to submit it!

**Comment by bschlenk** [comment]
I agree that stylistic rules exist to enforce consistency, but in this case it also enforces readability. `array.includes('value')` is much clearer than `array.indexOf('value') !== -1`. However, I don't find `/value/.test(str)` to be any less readable than `str.includes('value')`. Converting a regex text to the latter is going too far in my opinion, which is why an option to not touch regex at all would be great.

Modifying regex in this case is very unexpected given the title of the rule, "Enforce `includes` method over `indexOf` method". Using a regex to test a string is not the same as `indexOf`.

I'm not trying to come up with some definition of a simple regex that I like, it is taken directly from the incorrect examples in the doc:

```
// simple RegExp test
/foo/.test(str);
```

Would you be willing to reconsider? At the very least, the title of the rule should be updated to mention that it also will flag regex tests.
